### PR TITLE
[POAE7-2512] cast func numeric type support with arrow format

### PR DESCRIPTION
**Pull request by kd97**
### What changes were proposed in this pull request?

codegen cast func code refactor with arrow format.
### Why are the changes needed?

code refactor with arrow format migration.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

UTs
### Which label does this PR belong to?

CODEREFACTOR

### CODE REVIEW

File: cider/tests/functionality/CiderCastTest.cpp [hunk 1]
```diff
@@ -28,32 +30,36 @@
...
-    assertQuery("SELECT CAST(col_a as BIGINT), CAST(col_b as BIGINT) FROM test");       \
-    assertQuery("SELECT CAST(col_a as FLOAT), CAST(col_b as FLOAT) FROM test");         \
-    assertQuery("SELECT CAST(col_a as DOUBLE), CAST(col_b as DOUBLE) FROM test");       \
-    assertQuery(                                                                        \
-        "SELECT CAST(col_a as DOUBLE)  FROM test where CAST(col_b as INTEGER) > 20 ");  \
-    assertQuery("SELECT CAST(col_a as INTEGER), count(col_b) FROM test GROUP BY col_a", \
-                "",                                                                     \
-                true);                                                                  \
-    assertQuery("SELECT CAST(col_a as INTEGER) + CAST(col_b as INTEGER) FROM test");    \
+#define TEST_UNIT(TEST_CLASS, UNIT_NAME)                                                 \
```
Comment (jikunshang) [inline cider/tests/functionality/CiderCastTest.cpp:47]: I think it's better to keep original test here.

File: cider/exec/template/CastIR.cpp [hunk 2]
```diff
@@ -58,21 +58,24 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCastFun(
...
+  auto operand_nullable = codegen(operand, co, true);
+  auto operand_fixedsize = dynamic_cast<FixedSizeColValues*>(operand_nullable.get());
+  CHECK(operand_fixedsize);
+  auto operand_lv = operand_fixedsize->getValue();
+  llvm::Value* target_lv = nullptr;
 
-  return std::make_unique<FixedSizeColValues>(cast_lv, fixedsize_lv->getNull());
+  if (operand_lv->getType()->isIntegerTy()) {
+    if (ti.is_integer() || ti.is_decimal() || ti.is_boolean()) {
+      target_lv = codegenCastBetweenIntTypesForArrow(operand_lv, operand_ti, ti);
```
Comment (jikunshang) [inline cider/exec/template/CastIR.cpp:71]: will (cast int32 to int64) fall into this branch?

File: cider/exec/template/CodeGenerator.h [hunk 3]
```diff
@@ -418,14 +424,24 @@ class CodeGenerator {
                                      const bool operand_is_const,
                                      const CompilationOptions& co);
 
+  // to be deprecated
   llvm::Value* codegenCastToFp(llvm::Value* operand_lv,
                                const SQLTypeInfo& operand_ti,
                                const SQLTypeInfo& ti);
 
+  llvm::Value* codegenCastToFpForArrow(llvm::Value* operand_lv,
```
Comment (yma11) [inline cider/exec/template/CodeGenerator.h:432]: What's the main difference between original `codegenCastToFp` and new `codegenCastToFpForArrow`?

File: cider/exec/template/CastIR.cpp (hunk 2 above)
Comment (kd97) [reply cider/exec/template/CastIR.cpp]: `is_integer()` including tinyint,smallint,int and bigint.

File: cider/exec/template/CodeGenerator.h (hunk 3 above)
Comment (kd97) [reply cider/exec/template/CodeGenerator.h:438]: Removed null value check and simplified type conversion code.

File: cider/tests/functionality/CiderFilterTest.cpp [hunk 4]
```diff
@@ -433,14 +435,11 @@ TEST_F(CiderProjectAllTestArrow, ArrowFilterProjectAllTest) {
   assertQueryArrow("SELECT * FROM test where TRUE");
   assertQueryArrow(
       "SELECT * FROM test where (col_3 > 0 and col_4 > 0) or (col_5 < 0 and col_6 < 0) ");
-
-  // TODO(yizhong): Enable this after date and bool is supported in arrow.
-  GTEST_SKIP_("date codegen is not ready.");
-  assertQueryArrow("SELECT * FROM test where col_2 <> 0 and col_7 > '1972-02-01'");
+  assertQueryArrow("SELECT * FROM test where col_2 <> 0 and col_7 > date '1972-02-01'");
```
Comment (jikunshang) [inline cider/tests/functionality/CiderFilterTest.cpp:438]: can we add a query for bool type here?

**Review by haiweizh — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**